Modules/ManualAssessment/classes/class.ilManualAssessmentMemberGUI.php

```diff
@@ -87,10 +89,21 @@ protected function edit($form = null)
 		}
 		$form->addCommandButton('save', $this->lng->txt('save'));
 		$form->addCommandButton('finalizeConfirmation', $this->lng->txt('mass_finalize'));
+		$form = $this->possiblyAddDownloadAttachemnentButtonTo($form);
 		$form->addCommandButton('cancel', $this->lng->txt('mass_return'));
 		$this->renderForm($form);
 	}
 
+	protected function downloadAttachement()
+	{
+		if (!$this->mayBeEdited() && !$this->mayBeViewed() && !$this->mayBeAmended()) {
+			$this->parent_gui->handleAccessViolation();
```

Comment:
Vllt. hier noch nen `return` hinter, sonst ändere ich in handleAccessViolation was und komme plötzlich doch wieder an die Datei.